pkg/controller/vitesskeyspace/reconcile_shards.go

```diff
@@ -187,6 +204,45 @@ func allShardsServingWrites(shardNames []string, shardStatus map[string]planetsc
 	return result
 }
 
+func totalReadyTablets(shardNames []string, shardStatus map[string]planetscalev2.VitessKeyspaceShardStatus) int32 {
+	var tabletCount int32
+	for _, shard := range shardNames {
+		tabletCount += shardStatus[shard].ReadyTablets
+	}
+
+	return tabletCount
+}
+
+func totalTablets(shardNames []string, shardStatus map[string]planetscalev2.VitessKeyspaceShardStatus) int32 {
+	var tabletCount int32
+	for _, shard := range shardNames {
+		tabletCount += shardStatus[shard].Tablets
+	}
+
+	return tabletCount
+}
+
+func totalUpdatedTablets(shardNames []string, shardStatus map[string]planetscalev2.VitessKeyspaceShardStatus) int32 {
+	var tabletCount int32
+	for _, shard := range shardNames {
+		tabletCount += shardStatus[shard].UpdatedTablets
+	}
+
+	return tabletCount
+}
+
+func totalReadyShards(shardNames []string, shardStatus map[string]planetscalev2.VitessKeyspaceShardStatus) int32 {
+	var readyShards int32
+	for _, shardName := range shardNames {
+		shard := shardStatus[shardName]
+		if shard.ReadyTablets == shard.Tablets {
```

Comment:
I think we should look at DesiredTablets so we don't think everything is fine if a tablet is missing. It looks like I got this wrong myself in `vitesscluster/reconcile_keyspaces.go`. Can you fix it there too?

```suggestion
		if shard.ReadyTablets == shard.DesiredTablets {
```

Reply:
Good point. I'll fix it in both spots!

Reply:
Done.